src/FieldValidator.php

```diff
@@ -148,4 +149,63 @@ private function assertValidType(string $type, $value): bool
     {
         return $value instanceof $type || gettype($value) === $type;
     }
+
+    protected function resolveAllowedArrayTypesFromCollection(string $type): array
+    {
+        if (! $type || ! class_exists($type) || ! is_subclass_of($type, DataTransferObjectCollection::class)) {
```

Comment:
Wouldn't `! is_subclass_of($type, DataTransferObjectCollection::class)` suffice? It'd also return false for the other conditions I believe.